# Inspection et exploration du dataset

In [ ]:
import seaborn as sns

In [ ]:
tips = sns.load_dataset("tips")

## Structure du dataset

In [3]:
tips.shape

(244, 7)

**Constat**:

- 7 variables et 244 lignes de données

In [4]:
tips.head(10)

,total_bill,tip,sex,smoker,day,time,size
0,16.99,1.01,Female,No,Sun,Dinner,2
1,10.34,1.66,Male,No,Sun,Dinner,3
2,21.01,3.50,Male,No,Sun,Dinner,3
3,23.68,3.31,Male,No,Sun,Dinner,2
4,24.59,3.61,Female,No,Sun,Dinner,4
5,25.29,4.71,Male,No,Sun,Dinner,4
6,8.77,2.00,Male,No,Sun,Dinner,2
7,26.88,3.12,Male,No,Sun,Dinner,4
8,15.04,1.96,Male,No,Sun,Dinner,2
9,14.78,3.23,Male,No,Sun,Dinner,2


In [5]:
tips.tail(10)

,total_bill,tip,sex,smoker,day,time,size
234,15.53,3.00,Male,Yes,Sat,Dinner,2
235,10.07,1.25,Male,No,Sat,Dinner,2
236,12.60,1.00,Male,Yes,Sat,Dinner,2
237,32.83,1.17,Male,Yes,Sat,Dinner,2
238,35.83,4.67,Female,No,Sat,Dinner,3
239,29.03,5.92,Male,No,Sat,Dinner,3
240,27.18,2.00,Female,Yes,Sat,Dinner,2
241,22.67,2.00,Male,Yes,Sat,Dinner,2
242,17.82,1.75,Male,No,Sat,Dinner,2
243,18.78,3.00,Female,No,Thur,Dinner,2


In [6]:
tips.info()

<class 'pandas.DataFrame'>
RangeIndex: 244 entries, 0 to 243
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype   
---  ------      --------------  -----   
 0   total_bill  244 non-null    float64 
 1   tip         244 non-null    float64 
 2   sex         244 non-null    category
 3   smoker      244 non-null    category
 4   day         244 non-null    category
 5   time        244 non-null    category
 6   size        244 non-null    int64   
dtypes: category(4), float64(2), int64(1)
memory usage: 7.4 KB


**Constat**:

- Pas de colonne d'identifiant

- Aucune valeur manquante à première vue

- 3 variables numériques et 4 variables catégorielles

## Types de données

In [7]:
tips.dtypes

total_bill     float64
tip            float64
sex           category
smoker        category
day           category
time          category
size             int64
dtype: object

### Classification des variables

- `total_bill` et `tip` sont des variables quantitatives continues

- `size` est une variable quantitative discrète

In [8]:
print(tips["sex"].unique())
print(tips["day"].unique())
print(tips["time"].unique())
print(tips["smoker"].unique())

['Female', 'Male']
Categories (2, str): ['Male', 'Female']
['Sun', 'Sat', 'Thur', 'Fri']
Categories (4, str): ['Thur', 'Fri', 'Sat', 'Sun']
['Dinner', 'Lunch']
Categories (2, str): ['Lunch', 'Dinner']
['No', 'Yes']
Categories (2, str): ['Yes', 'No']


**Constat**:
- `sex`, `day`, `time` et `smoker` sont des variables catégorielles

## Valeurs manquantes

In [9]:
tips.isna().mean()

total_bill    0.0
tip           0.0
sex           0.0
smoker        0.0
day           0.0
time          0.0
size          0.0
dtype: float64

In [10]:
tips.isna().sum()

total_bill    0
tip           0
sex           0
smoker        0
day           0
time          0
size          0
dtype: int64

**Constat** : Une moyenne de 0% de valeurs manquantes et une absence de lignes avec des valeurs manquantes dans le dataset

## Doublon

In [11]:
tips[tips.duplicated()]

,total_bill,tip,sex,smoker,day,time,size
202,13.0,2.0,Female,Yes,Thur,Lunch,2


**Constat** : 
- Présence d'un doublon sur la ligne 202

**Hypothèse**: 
- doublon probable de saisie, non vérifiable sans identifiant

In [12]:
tips[tips.duplicated(keep=False)]

,total_bill,tip,sex,smoker,day,time,size
198,13.0,2.0,Female,Yes,Thur,Lunch,2
202,13.0,2.0,Female,Yes,Thur,Lunch,2


### Décision

- Supprimer le doublon évite d'introduire un biais silencieux même si c'est négligeable avec un unique doublon

In [13]:
tips = tips.drop_duplicates()

**Vérification de la suppression**:

In [14]:
tips[tips.duplicated(keep=False)]

,total_bill,tip,sex,smoker,day,time,size


In [15]:
tips.value_counts().head()

total_bill  tip   sex     smoker  day  time    size
16.99       1.01  Female  No      Sun  Dinner  2       1
10.34       1.66  Male    No      Sun  Dinner  3       1
21.01       3.50  Male    No      Sun  Dinner  3       1
23.68       3.31  Male    No      Sun  Dinner  2       1
24.59       3.61  Female  No      Sun  Dinner  4       1
Name: count, dtype: int64